Source/Mockolate/Setup/ReturnMethodSetup.cs

```diff
@@ -236,6 +267,28 @@ public IReturnMethodSetupCallbackBuilder<TReturn, T1> Do(Action<int, T1> callbac
 		return this;
 	}
 
+	/// <summary>
+	///     Registers a <paramref name="callback" /> to execute when the method is called.
+	/// </summary>
+	public IReturnMethodSetupCallbackBuilder<TReturn, T1> Do<TMock>(Action<TMock, T1> callback)
+	{
+		Callback<Action<int, T1>> currentCallback = new((_, p1) => callback((TMock)Registration!.Subject!, p1));
+		_currentCallback = currentCallback;
+		_callbacks.Add(currentCallback);
+		return this;
+	}
+
+	/// <summary>
+	///     Registers a <paramref name="callback" /> to execute when the method is called.
+	/// </summary>
+	public IReturnMethodSetupCallbackBuilder<TReturn, T1> Do<TMock>(Action<TMock, int, T1> callback)
```

Comment:
The changes that were applied here should also be applied in the generated sources for `ReturnMethodSetup` in `Sources.MethodSetups.cs`.